thetis/utility.py

```diff
@@ -306,11 +306,64 @@ def comp_tracer_mass_2d(eta, bath, scalar_func):
     :arg bath: bathymetry :class:`Function`
     :arg scalar_func: scalar :class:`Function` to integrate
     """
-
     val = assemble((eta+bath)*scalar_func*dx)
     return val
 
 
+def comp_tracer_total_mass_2d(var, tracer_name):
```

Comment:
This is somewhat a strange way to define the total tracer mass computation. As it looks like this method cannot be called without the corresponding callback object (`var` argument) it would be better to move all of this inside that callback. Note that e.g. `comp_tracer_mass_2d` only depends on the fields, so it makes sense to have a separate utility function.